context/app/static/js/components/Header/UserLinks/UserLinks.jsx

```diff
@@ -10,13 +11,14 @@ import { TruncatedSpan, WarningDropdownLink } from './style';
 function UserLinks({ isAuthenticated, userEmail }) {
   const { isWorkspacesUser } = useAppContext();
 
+  const userType = getUserType();
   return (
     <Dropdown
       title={<TruncatedSpan>{isAuthenticated ? userEmail || 'User' : 'User Profile'}</TruncatedSpan>}
       data-testid="user-profile-dropdown"
     >
       <DropdownLink href="/my-lists">My Lists</DropdownLink>
-      {isWorkspacesUser && <DropdownLink href="/workspaces">My Workspaces</DropdownLink>}
+      {(isWorkspacesUser || userType === 'internal') && <DropdownLink href="/workspaces">My Workspaces</DropdownLink>}
```

Comment:
Perhaps adjusting the "isWorkspacesUser" logic in the app context instead could help avoid unexpected regressions in other places?

Reply:
We are only changing the behavior for the dropdown to be available to workspace users or users who have authenticated in the past (also covers currently authenticated, non-workspace users). The auth guard will behave the same as it has. Maybe I'm misinterpreting your comment.

Reply:
Added a var to make this more clear per your suggestion.